### PR TITLE
PyPI Release Pipeline

### PR DESCRIPTION
**Pull request by riedgar-ms**
Create an ADO pipeline for doing releases to PyPI, in the pattern of other projects in RAI.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -21,5 +21,6 @@ pytest-cov
 nbformat
 papermill
 nteract-scrapbook
-shap
-interpret-core
+gevent>=1.3.6
+interpret-core[required]>=0.1.20, <=0.1.21
+shap>=0.20.0, <=0.34.0
```
Comment (riedgar-ms) [inline requirements.txt:26]: These three are all taken from:
https://github.com/interpretml/interpret-community/blob/e82fc0904dd999f7aa59c943d956d1dff552108d/python/setup.py#L42

Comment (wamartin-aml) [reply requirements.txt:26]: yeah, this pin looks better than the interpret-text one, no issues form me

File: python/interpret_community/__init__.py [hunk 2]
```diff
@@ -32,7 +32,8 @@ def close_handler():
         logger.removeHandler(handler)
     atexit.register(close_handler)
 
+__name__ = "interpret_community"
 _major = '0'
 _minor = '14'
 _patch = '0'
-__version__ = '{}.{}.{}'.format(_major, _minor, _patch)
+__version__ = '{}.{}.{}.dev4'.format(_major, _minor, _patch)
```
Comment (wamartin-aml) [inline python/interpret_community/__init__.py:39]: Flagging just in case

File: devops/PyPI-Release.yml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    vmImage: $(poolImage)
+  
+  variables:
+    envInfoArtifact: TestPyPIFreeze
+    envInfoFileBase: requirements-pypi-test
+
+  jobs:
+  - template: templates/all-tests-job-template.yml
+    parameters:
+      platforms:  { Linux: ubuntu-latest, MacOS: macos-latest, Windows: windows-latest }
```
Comment (imatiach-msft) [inline devops/PyPI-Release.yml:193]: the macos tests work?  I thought there were some tolerance issues?

File: python/interpret_community/__init__.py (hunk 2 above)
Comment (riedgar-ms) [reply python/interpret_community/__init__.py]: Removed, thanks.

File: devops/PyPI-Release.yml (hunk 3 above)
Comment (riedgar-ms) [reply devops/PyPI-Release.yml:193]: They appear to be intermittent, unfortunately. We might want to trim the testing when it comes time to do an actual release. I have managed to get all stages to work, but when tests are unreliable, we will have problems.

**Pull request merged**